src/ert/config/queue_config.py

```diff
@@ -139,17 +113,139 @@ def generate_dict(option_list: List[Tuple[str, str]]) -> Dict[str, List[str]]:
+        "MAX_RUNTIME",
+    ],
+    "TORQUE": ["SUBMIT_SLEEP", "QUEUE_QUERY_TIMEOUT"],
+    "LOCAL": [],
+}
+
+queue_bool_options: Mapping[str, List[str]] = {
+    "LSF": ["DEBUG_OUTPUT"],
+    "SLURM": [],
+    "TORQUE": ["KEEP_QSUB_OUTPUT"],
+    "LOCAL": [],
+}
+
+
+def _validate_queue_driver_settings(
+    queue_system_options: List[Tuple[ContextString, ContextString]], queue_type: str
+) -> None:
+    for option_strings in queue_system_options:
         option_name = option_strings[0]
         option_value = option_strings[1]
+        if option_value == "":  # This is equivalent to the option not being set
+            continue
         if (
-            option_value != ""  # This is equivalent to the option not being set
-            and option_name == "MEMORY_PER_JOB"
+            option_name in queue_memory_options[queue_type]
             and re.match("[0-9]+[mg]b", option_value) is None
         ):
-            raise ConfigValidationError(
-                f"The value '{option_value}' is not valid for the Torque option "
-                "MEMORY_PER_JOB, it must be of "
-                "the format '<integer>mb' or '<integer>gb'."
+            raise ConfigValidationError.with_context(
+                f"'{option_value}' for {option_name} is not a valid format "
+                f"for {queue_type} QUEUE. It must be of "
+                "the format '<integer>mb' or '<integer>gb'.",
+                option_value,
+            )
+        if option_name in queue_string_options[queue_type] and not isinstance(
+            option_value, str
+        ):
+            raise ConfigValidationError.with_context(
+                f"'{option_value}' for {option_name} is not a valid string type "
+                f"for the {queue_type} QUEUE.",
+                option_value,
+            )
+        if option_name in queue_positive_number_options[queue_type] and (
+            re.match(r"^\d+(\.\d+)?$", option_value) is None
+        ):
+            raise ConfigValidationError.with_context(
+                f"'{option_value}' for {option_name} is not a valid integer or float"
+                f" for the {queue_type} QUEUE.",
+                option_value,
+            )
+        if option_name in queue_positive_int_options[queue_type] and (
+            re.match(r"^\d+$", option_value) is None
+        ):
+            raise ConfigValidationError.with_context(
+                f"'{option_value}' for {option_name} is not a valid positive integer"
+                f" for the {queue_type} QUEUE.",
+                option_value,
+            )
+        if option_name in queue_bool_options[queue_type] and not option_value in [
+            "TRUE",
+            "FALSE",
+            "0",
+            "1",
+            "T",
+            "F",
+        ]:
+            raise ConfigValidationError.with_context(
+                f"The '{option_value}' for {option_name} must be one of TRUE, FALSE, T, F, 1, 0 "
+                f" for the {queue_type}.",
```

Comment:
This is probably independent of queue type?

Reply:
Not really. There are some, which might be different for each type; eg. DEBUG_OUTPUT